src/libraries/System.Private.CoreLib/src/System/Convert.cs

```diff
@@ -2446,71 +2452,91 @@ public static unsafe bool TryToBase64Chars(ReadOnlySpan<byte> bytes, Span<char>
                 {
-                    if (insertLineBreaks)
+                    var rounds = insertLineBreaks
+                        ? Math.Min(calcLength, base64LineBreakPosition / 4 * 3)
+                        : calcLength;
+
+                    calcLength -= rounds;
+                    // Convert three bytes at a time to base64 notation.  This will consume 4 chars.
+                    for (; rounds > 0; rounds -= 3)
                     {
-                        if (charcount == base64LineBreakPosition)
-                        {
-                            outChars[j++] = '\r';
-                            outChars[j++] = '\n';
-                            charcount = 0;
-                        }
-                        charcount += 4;
+                        a = *inData++;
+                        b = *inData++;
+                        c = *inData++;
+                        *outPairs++ = base64Pairs[(a << 4) | (b >> 4)];
+                        *outPairs++ = base64Pairs[((b << 8) | c) & 0xfff];
                     }
-                    outChars[j] = base64[(inData[i] & 0xfc) >> 2];
-                    outChars[j + 1] = base64[((inData[i] & 0x03) << 4) | ((inData[i + 1] & 0xf0) >> 4)];
-                    outChars[j + 2] = base64[((inData[i + 1] & 0x0f) << 2) | ((inData[i + 2] & 0xc0) >> 6)];
-                    outChars[j + 3] = base64[inData[i + 2] & 0x3f];
-                    j += 4;
-                }
 
-                // Where we left off before
-                i = calcLength;
-
-                if (insertLineBreaks && (lengthmod3 != 0) && (charcount == base64LineBreakPosition))
+                    if (insertLineBreaks
+                        && calcLength > 0
+                        && ((char*)outPairs - outChars) % (base64LineBreakPosition + 2) == base64LineBreakPosition)
+                    {
+                        *outPairs++ = crlf;
+                    }
+                }
+                if (insertLineBreaks
+                    && lengthmod3 != 0
+                    && ((char*)outPairs - outChars) % (base64LineBreakPosition + 2) == base64LineBreakPosition)
                 {
-                    outChars[j++] = '\r';
-                    outChars[j++] = '\n';
+                    *outPairs++ = crlf;
                 }
 
+                char* outPtr;
                 switch (lengthmod3)
                 {
                     case 2: // One character padding needed
-                        outChars[j] = base64[(inData[i] & 0xfc) >> 2];
-                        outChars[j + 1] = base64[((inData[i] & 0x03) << 4) | ((inData[i + 1] & 0xf0) >> 4)];
-                        outChars[j + 2] = base64[(inData[i + 1] & 0x0f) << 2];
-                        outChars[j + 3] = base64[64]; // Pad
-                        j += 4;
+                        a = *inData++;
+                        b = *inData++;
+                        *outPairs++ = base64Pairs[(a << 4) | (b >> 4)];
```

Comment:
@tannergooding 
Why is the codegen soo different to LLVM? https://godbolt.org/z/T8eqE9chv

### LLVM
```asm
        shr     sil, 4
        shl     dil, 4
        or      dil, sil
        movzx   eax, dil
        ret
```

### .NET JIT
```asm
       movzx    rax, dil
       shl      eax, 4
       movzx    rdi, sil
       sar      edi, 4
       or       eax, edi
       ret      
```

Reply:
@deeprobin feel free to file an issue. Related one is https://github.com/dotnet/runtime/issues/13816